packages/kit/src/module/expose.ts

```diff
@@ -0,0 +1,44 @@
+import { resolve } from 'pathe'
+import { useNuxt } from '../context'
+import { addTemplate } from '../template'
+
+/**
+ * Expose virtual module config to the NuxtApp and nitro with TypeScript support.
+ */
+export function exposeRuntimeModuleConfig (moduleName: string, config: Record<string, any>) {
+  const nuxt = useNuxt()
+  const jsExports = Object.entries(config).map(([k, v]) => `export const ${k} = ${JSON.stringify(v)}`).join('\n')
+  const alias = `#${moduleName}/config`
+  nuxt.options.alias[alias] = addTemplate({
+    filename: `modules/config/${moduleName}.mjs`,
+    getContents: () => jsExports
+  }).dst
+  // expose for nitro
+  nuxt.hooks.hook('nitro:config', (nitroConfig) => {
+    nitroConfig.virtual![alias] = jsExports
+  })
+
+  // need to create a ts def for the file
+  const typeDefName = `modules/config/${moduleName}.d.ts`
+  const tsExports = Object.keys(config).map((k) => {
+    // expose actual values in dev
+    if (nuxt.options.dev) { return `  export const ${k}: ${JSON.stringify(config[k])} | ModuleOptions['${k}']` }
+    return `  export const ${k}: ModuleOptions['${k}']`
```

Comment:
What if the options exposed by the module aren't similar to `ModuleOptions[k]` (i.e. they were transformed, parsed, resolved)? Not sure if this is out of scope for this functions and meant for a lower level alternative :thinking: